tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/statements/vardeclr/ModuleVariableTest.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.ballerinalang.test.statements.vardeclr;
+
+import org.ballerinalang.core.model.values.BBoolean;
+import org.ballerinalang.core.model.values.BByte;
+import org.ballerinalang.core.model.values.BFloat;
+import org.ballerinalang.core.model.values.BInteger;
+import org.ballerinalang.core.model.values.BValue;
+import org.ballerinalang.test.BCompileUtil;
+import org.ballerinalang.test.BRunUtil;
+import org.ballerinalang.test.CompileResult;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static org.ballerinalang.test.BAssertUtil.validateError;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Class to test module level variable declaration for all binding patterns.
+ *
+ * @since 2.0
+ */
+public class ModuleVariableTest {
+
+    private CompileResult compileResult, compileResultNegative;
+
+    @BeforeClass
+    public void setup() {
+        compileResult = BCompileUtil.compile("test-src/statements/vardeclr/module_tuple_var_decl.bal");
+        compileResultNegative = BCompileUtil.compile("test-src/statements/vardeclr/module_tuple_var_decl_negetive.bal");
+    }
+
+    @Test
+    public void testBasicModuleLevelTupleVarDecl() {
+
+        BValue[] returns = BRunUtil.invoke(compileResult, "testBasic");
+
+        int index = 0;
+        assertTrue(((BBoolean) returns[index++]).booleanValue());
```

Comment:
Shall we write these assertions in ballerina itself?

e.g., https://github.com/ballerina-platform/ballerina-lang/blob/v2.0.0-Preview5/tests/jballerina-unit-test/src/test/resources/test-src/types/readonly/test_selectively_immutable_type.bal#L990

Reply:
Done